analyses/copy_number_consensus_call/src/scripts/get_rid_bad_segments.py

```diff
@@ -0,0 +1,135 @@
+                                             centromeric, and seg-dup regions""")
+parser.add_argument('--reference', required=True,
+                    help='path to the reference list of bad segments to be filtered out for')
+parser.add_argument('--file', required=True,
+                    help='path to the CNVs file that needs to be filtered')
+
+args = parser.parse_args()
+
+#querie_file = sys.argv[2]
+
+#db_file = sys.argv[1]    #bad segment list
+
+
+## Check to see if the CNVs file is empty, if so, make an empty variable
+if os.stat(args.file).st_size == 0:
+    final_content = []
+
+## If the file has content, then load in the content
+else:
+    with open (args.file) as file:
+
+        ## Strip the trailing spaces for each line
+        stripped_content = [i.rstrip() for i in file.readlines()]
+
+        ## If the file is TAB delimited, split the file up by TABs
+        if stripped_content[0].find('\t') != -1:
+            final_content = [i.split('\t') for i in stripped_content]
+
+        ## If the file is space delimited, split the file up by the spaces
+        else:
+            final_content = [i.split() for i in stripped_content]
+
+        ## If the 1st column has '1' instead of 'chr1' for the chromosome number, add in 'chr'
+        if final_content[0][0].find('chr') == -1:
+            for c,chromo in enumerate(final_content):
+                final_content[c][0] = 'chr' + final_content[c][0]
+
+
+
+## Works with the assumption that the segments in the file on the next line DO NOT OVERLAP
+## Read in the reference file
+with open(args.reference) as file:
+
+    ## Strip the trailing spaces for each line
+    reference_stripped = [i.rstrip() for i in file.readlines()]
+
+    ## If the file is TAB delimited, split the file up by TABs
+    if reference_stripped[0].find('\t') != 1:
+        fin_reference_file = [i.split('\t') for i in reference_stripped]
+
+    ## If the file is space delimited, split the file up by the spaces
+    else:
+        fin_reference_file = [i.split() for i in reference_stripped]
+
+    ## If the 1st column has '1' instead of 'chr1' for the chromosome number, add in 'chr'
+    if fin_reference_file[0][0].find('chr') == -1:
+        for c,chromo in enumerate(fin_reference_file):
+            fin_reference_file[c][0] = 'chr' + fin_reference_file[c][0]
+
+
+## Define a variable to hold the final results
+fin_list = []
+
+## Loop through the CNVs file.
+## If any segment in the reference file overlap >50%, then get rid of that CNV
+for i in final_content:
+
+    ## Initiate a variable for the total overlapping
+    overlap = 0
+
+    ## Get the start/end coordinates of the cnv
+    start_cnv = int(i[1])
+    end_cnv = int(i[2])
+
+    ## Make sure end > start (end downstream from start)
+    if start_cnv > end_cnv:
+        start_cnv, end_cnv = end_cnv, start_cnv
+
+    ## For each CNV, loop through the entire reference file
```

Comment:
This step could make things slow. Are the CNVs sorted? If so, we can probably keep track of where we are in the reference file.  If the CNV is after our current reference position, we can search forward in the reference, but we won't need to go all the way to the end, and we won't need to go back to the start for the next CNV. I don't know how long this takes on the full data, but if it is slow, this might be a place to look for a speed boost.